File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.nextLine());
        int[] arr = Arrays.stream(sc.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int target = Integer.parseInt(sc.nextLine());
        
        System.out.print(countPairs(arr, target));
        sc.close();
    }
    
    private static int countPairs(int[] arr, int target) {
        Set<Integer> seen = new HashSet<>();
        int count = 0;
        
        for (int num : arr) {
            int complement = target - num;
            if (seen.contains(complement)) {
                count++;
            }
            seen.add(num);
        }
        return count;
    }
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
import sys
read_line = sys.stdin.readline

def count_pairs(arr, target):
    seen = {}
    count = 0
    
    for num in arr:
        complement = target - num
        if complement in seen:
            count += 1
        seen[num] = True
    
    return count

n = int(read_line())
arr = list(map(int, read_line().split()))
target = int(read_line())
print(count_pairs(arr, target))